modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -168,6 +183,19 @@ public class DistributionZoneManager implements IgniteComponent {
      */
     private final Map<Integer, ZoneState> zonesState;
 
+    /** The tracker for last topology version which was observed by distribution zone manager. */
+    private final PendingComparableValuesTracker<Long> topVerTracker;
+
+    /**
+     * The last meta storage revision on which scale up timer was started.
+     */
+    private long lastScaleUpRevision;
```

Comment:
AFAIK we need volatile here(and also for lastScaleDownRevision), because otherwise you might miss lastScale<>Revision updates triggered from within onUpdateScaleUp()

Reply:
Fixed